aienvs/Sumo/SumoHelper.py

```diff
@@ -125,6 +122,65 @@ def _generate_route_file(self, seed):
         Generates vehicles for each possible route in the scenario and writes
         them to file. Returns the location of the sumocfg file.
         """
+
+        valid_route_generation_methods = ['legacy', 'randomTrips.py', 'activitygen']
+
+        route_name = str(self._port) + '_routes.rou.xml'
+        route_file = os.path.join(self.scenario_path, route_name)
+        net_file = os.path.join(self.scenario_path, self._net_file)
+
+        assert self.parameters['route_generation_method'] in valid_route_generation_methods
+
+        if self.parameters['route_generation_method'] == 'randomTrips.py':
+            logging.debug('Using sumo/tools/randomTrips.py to generate trips')
+            params = ['-n', net_file, '-o', route_file]
+            if seed is not None:
+                params += ['-s', str(seed)]
+
+            randomTrips.main(randomTrips.get_options(self.parameters['trips_generate_options'] + params))
+
+        elif self.parameters['route_generation_method'] == 'activitygen':
+            logging.debug('Using activitygen and duarouter to generate trips based on stat-file')
+
+            stat_file = get_stat_file(scenario_path=self.scenario_path)
+
+            # Used as input to duarouter
+            tmp_trips_route_name = str(self._port) + '_routes.trips.rou.xml'
+            tmp_trips_route_file = os.path.join(self.scenario_path, route_name)
+
+            # Get the paths of the sumotools binaries activitygen and duarouter
+            ACTIVITYGEN = sumolib.checkBinary('activitygen')
+            DUAROUTER = sumolib.checkBinary('duarouter')
+
+            activitygen_args = [ACTIVITYGEN, '--net-file', net_file, '--stat-file', stat_file,
+                                '--output-file', tmp_trips_route_file]
+
+            # Add passed through arguments
+            activitygen_args += self.parameters['activitygen_options']
+
+            if seed is not None:
+                activitygen_args += ['-s', str(seed)]
+
+            subprocess.call(activitygen_args)
+
+            duarouter_args = [DUAROUTER, '--net-file', net_file, '--route-files', tmp_trips_route_file,
```

Comment:
The random trips method only generates trips instead of also routing the cars. Only calling activitygen and not also duarouter would give the same behaviour.

If only trips are supplied to sumo, [routes are calculated when a vehicle departs](https://sumo.dlr.de/docs/Demand/Dynamic_User_Assignment.html#oneshot-assignment) taking traffic jams into account, so it's probably a good idea to also only supply trips for the activitygen method.

Reply:
I changed it to using trips directly.